Makefile

```diff
@@ -159,19 +159,16 @@ openapi/spec/validate: specinstall
 	${SPECTRAL} lint ../../openapi/kas-fleet-manager.yaml ../../openapi/kas-fleet-manager-private-admin.yaml ;\
 	}
 
-	
-
-ifeq ($(shell uname -s | tr A-Z a-z), darwin)
-        PGHOST:="127.0.0.1"
-else
-        PGHOST:="172.18.0.22"
-endif
-
 ### Environment-sourced variables with defaults
 # Can be overriden by setting environment var before running
 # Example:
 #   OCM_ENV=testing make run
 #   export OCM_ENV=testing; make run
+
+ifeq ($(shell uname -s | tr A-Z a-z), darwin)
+    CONTAINER_IMAGE_BUILD_PLATFORM ?= --platform linux/amd64
```

Comment:
I see some issues on the CI side.
Maybe this default value has to be quoted?

It contains dashes which mght be being interpreted as some shell flag

Reply:
i tried removing this assignment and i'm also getting the same issue. 

In main, as soon as I removed these lines, I get the same issue as what's shown in the CI
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/738b7b881aa293915f8374e1ae380549f75e5d8c/Makefile#L165-L169

These vars never get assigned which causes the error
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/738b7b881aa293915f8374e1ae380549f75e5d8c/Makefile#L177-L198

Going to look into this a bit further 👀

Reply:
@JameelB I'd suggest we defer removing this onto another PR.

Reply:
I see the spaces inside the PGHOST ifelse  are whitespaces and not tabs. Might that be causing the issue?

Reply:
> When testing this, it seems like Makefile doesn't like var definitions between target definitions.
> 
> I've moved all of the var definitions before the target definitions starts to ensure they're set correctly.

Even though this makes the tests pass and "fix the issue" it would be good to figure out what is happening.
A way to do this is to create a very small makefile where we try the combinations.

Reply:
Here's a gist with different combinations: https://gist.github.com/JameelB/4ecd37efb2cf4b88e7f75c6a1302e1c4

Reply:
I think I can identify where is the issue:

Inside `ifndef` ... `endif` blocks it seems that the characters used to indent matters.
For example, for the TEST_VAR_4 ifndef block:
if inside the body the TEST_VAR_4 initialization is not indented then it seems to work
if inside the body the TEST_VAR_4 initialization is indented with spaces it seems to work
if inside the body the TEST_VAR_4 initialization is indented with tabs it does not seem to work

Reply:
This means that the order where var definitions appear shouldn't really matter. The issue is another and has to do with indentation inside Makefile conditional directives

Reply:
good spot, that looks like it.

Vars like `OCM_ENV` which is used across multiple make targets should  probably stay at the top though. 

We can re-arrange some of the target specific ones like `TEST_SUMMARY_FORMAT` to be grouped with the test targets if preferred. Some of the vars here are only ever used for one target as well (i.e. SERVER_URL), in that case it should probably be defined as a target variable. 🤔 

There's probably more vars in there that needs to be reviewed to be removed or moved 🤔 I can create a follow on PR for this?

Reply:
rearranging them is good for organization purposes so we can do that.
But in any case we have to fix the issues of tabs being used in conditionals that are not shell conditionals but makefile conditionals.
For this PR let's do the order reorganization and fixing the indentation for the makefile conditionals

Reply:
i've updated this to remove the indentation from the make if statements. 

I've also moved the ones in this block to group them with their related make targets but I haven't looked at any of the other var definitions outside of this.